server/src/main/java/org/elasticsearch/search/SearchPhaseResult.java

```diff
@@ -90,6 +94,23 @@ public QuerySearchResult queryResult() {
      */
     public FetchSearchResult fetchResult() { return null; }
 
+    @Nullable
+    public ShardSearchRequest getShardSearchRequest() {
+        return shardSearchRequest;
+    }
+
+    public void setShardSearchRequest(ShardSearchRequest shardSearchRequest) {
+        this.shardSearchRequest = shardSearchRequest;
+    }
+
+    public RescoreDocIds getRescoreDocIds() {
+        return rescoreDocIds;
+    }
+
+    public void setRescoreDocIds(RescoreDocIds rescoreDocIds) {
+        this.rescoreDocIds = rescoreDocIds;
+    }
+
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         // TODO: this seems wrong, SearchPhaseResult should have a writeTo?
```

Comment:
we could apply the TODO here and clean the (de)serialization ? This would mean reconciling the logic between the different implementations and move the serialization of the object accessible here to this function. Happy to do this in a follow up though.

Reply:
++. I will do this in a follow-up.